test/Dispatcher.multiclient.sol

```diff
@@ -0,0 +1,104 @@
+        emit ChannelOpenTry(address(mars), "1.0", ChannelOrder.NONE, false, connectionHops0, ch1.portId, ch1.channelId);
+        dispatcherProxy.channelOpenTry(mars, ch0, ChannelOrder.NONE, false, connectionHops0, ch1, dummyProof);
+
+        // Having a dummy light client shouldn't impact opLightClient's ability to reject invalid proofs
+        vm.expectRevert(abi.encodeWithSelector(ProofVerifier.InvalidProofValue.selector));
+        dispatcherProxy.channelOpenTry(mars, ch1, ChannelOrder.NONE, false, connectionHops1, ch0, dummyProof);
+    }
+
+    function test_sendPacket_multiClient() public {
+        Ics23Proof memory openChannelProof = load_proof("/test/payload/channel_confirm_pending_proof.hex");
+        Ics23Proof memory invalidProof = load_proof("/test/payload/packet_ack_proof.hex");
+
+        // Set up valid connections from both dummy and real clients
+        dispatcherProxy.channelOpenConfirm(mars, ch0, connectionHops0, ChannelOrder.NONE, false, ch1, invalidProof);
+        dispatcherProxy.channelOpenConfirm(mars, ch1, connectionHops1, ChannelOrder.NONE, false, ch0, openChannelProof);
+
+        // dispatcherproxy.sendpacket should revert using channel 1
+        IbcEndpoint memory src = IbcEndpoint(IbcUtils.addressToPortId("polyibc.eth1.", address(mars)), ch1.channelId);
+        IbcEndpoint memory dest = IbcEndpoint(ch0.portId, ch0.channelId);
+
+        IbcPacket memory packet = IbcPacket(src, dest, uint64(1), bytes("sendPacket"), Height(0, 0), 0);
+        _storeSendPacketCommitment(address(mars), packet.src.channelId, 1, 1);
+        bytes memory ackData = bytes.concat(keccak256(hex"22726573756c7422"));
+
+        // OpProofverifier will cause acknowledgement to fail
+        vm.expectRevert(abi.encodeWithSelector(ProofVerifier.InvalidProofKey.selector));
+        dispatcherProxy.acknowledgement(mars, packet, ackData, invalidProof);
+
+        // Now we change the client to the dummy client and the packet should go through since it circumvents the proof
+        dispatcherProxy.setNewConnection(connectionHops1[0], dummyLightClient);
+        dispatcherProxy.acknowledgement(mars, packet, ackData, invalidProof);
+    }
+
+    function test_Dispatcher_Prevents_nonOwner_SetConnection() public {
+        vm.startPrank(notOwner);
+        vm.expectRevert("Ownable: caller is not the owner");
+        dispatcherProxy.setNewConnection("malicious-connection-1", opLightClient);
+    }
+
+    function test_Dispatcher_Prevents_nonOwner_RemoveConnection() public {
+        vm.startPrank(notOwner);
+        vm.expectRevert("Ownable: caller is not the owner");
+        dispatcherProxy.removeConnection(connectionHops0[0]);
+    }
+
+    function test_addr0_channels_cannot_be_added() public {
+        vm.expectRevert(abi.encodeWithSelector(IBCErrors.invalidAddress.selector));
+        dispatcherProxy.setNewConnection(connectionHops0[0], LightClient(address(0)));
+    }
+
+    function test_Dispatcher_removeConnection() public {
```

Comment:
ada a similar test for update an existing connection so it maps to a new client?

Reply:
this test should be covered [here](https://github.com/open-ibc/vibc-core-smart-contracts/blob/7bcedf7e670412a8bf97e06eddeb6d65fd75dcfc/test/Dispatcher.multiclient.sol#L67-L69) no?